packages/wdio-browserstack-service/src/Percy/Percy.ts

```diff
@@ -0,0 +1,172 @@
+import fs from 'node:fs'
+import path from 'node:path'
+import os from 'node:os'
+
+import { spawn } from 'node:child_process'
+
+import { nodeRequest, getBrowserStackUser, getBrowserStackKey, sleep } from '../util.js'
+import { PercyLogger } from './PercyLogger.js'
+
+import PercyBinary from './PercyBinary.js'
+
+import type { BrowserstackConfig, UserConfig } from '../types.js'
+import type { Options } from '@wdio/types'
+
+const logDir = 'logs'
+
+class Percy {
+    #logfile: string = path.join(logDir, 'percy.log')
+    #address: string = process.env.PERCY_SERVER_ADDRESS || 'http://127.0.0.1:5338'
+
+    #binaryPath: string | any = null
+    #options: BrowserstackConfig & Options.Testrunner
+    #config: Options.Testrunner
+    #proc: any = null
+    #isApp: boolean = false
+    #projectName: string | undefined = undefined
+
+    isProcessRunning = false
+
+    constructor(options: BrowserstackConfig & Options.Testrunner, config: Options.Testrunner, bsConfig: UserConfig) {
+        this.#options = options
+        this.#config = config
+        if (options.app) {
+            this.#isApp = true
+        }
+        this.#projectName = bsConfig.projectName
+    }
+
+    async #getBinaryPath(): Promise<string> {
+        if (!this.#binaryPath) {
+            const pb = new PercyBinary()
+            this.#binaryPath = await pb.getBinaryPath(this.#config)
+        }
+        return this.#binaryPath
+    }
+
+    async healthcheck() {
+        try {
+            const resp = await nodeRequest('GET', 'percy/healthcheck', null, this.#address)
+            if (resp) {
+                return true
+            }
+        } catch (err) {
+            return false
+        }
+    }
+
+    async start() {
+        const binaryPath: string = await this.#getBinaryPath()
+        const logStream = fs.createWriteStream(this.#logfile, { flags: 'a' })
+        const token = await this.fetchPercyToken()
+        const configPath = await this.createPercyConfig()
+
+        if (!token) {
+            return false
+        }
+
+        const commandArgs = [`${this.#isApp ? 'app:exec' : 'exec'}:start`]
+
+        if (configPath) {
+            commandArgs.push('-c', configPath as string)
+        }
+
+        this.#proc = spawn(
+            binaryPath,
+            commandArgs,
+            { env: Object.assign(process.env, { PERCY_TOKEN: token }) }
```

Comment:
Calling `Object.assign` will persist the token in the environment which you probably won't want, better:
```suggestion
            { env: { ...process.env, PERCY_TOKEN: token } }
```

Reply:
Done